code/modules/xenobio2/machinery/gene_manipulators.dm

```diff
@@ -0,0 +1,429 @@
+/*
+	This file contains:
+	
+	Xenobiological disk:
+	Holds traits that can be taken from cores and transplanted into slimes.
+	
+	Biological Product Destructive Analyzer:
+	Takes certain traits in gene grouping from a core and places them into a disk.
+	
+	Biological genetic bombarder:
+	Takes traits from a disk and replaces/adds to the genes in a xenobiological creature.
+*/
+/obj/item/weapon/disk/xenobio
+	name = "biological data disk"
+	desc = "A small disk used for carrying data on genetics."
+	icon = 'icons/obj/hydroponics_machines.dmi'
+	icon_state = "disk"
+	w_class = 1.0
+	
+	var/list/genes = list()
+	var/genesource = "unknown"
+	
+/obj/item/weapon/disk/xenobio/attack_self(var/mob/user as mob)
+	if(genes.len)
+		var/choice = alert(user, "Are you sure you want to wipe the disk?", "Xenobiological Data", "No", "Yes")
+		if(src && user && genes && choice && choice == "Yes" && user.Adjacent(get_turf(src)))
+			user << "You wipe the disk data."
+			name = initial(name)
+			desc = initial(name)
+			genes = list()
+			genesource = "unknown"
+			
+/obj/item/weapon/storage/box/xenobiodisk
+	name = "biological disk box"
+	desc = "A box of biological data disks, apparently."
+
+/obj/item/weapon/storage/box/xenobiodisk/New()
+	..()
+	for(var/i = 0 to 7)
```

Comment:
That'll spawn 8 disks into the box. Not sure if intentional, standard stuff seems to be 7.